internal/mux/server.go

```diff
@@ -100,6 +109,21 @@ func registerMetrics(httpServer *http.ServeMux) {
 	httpServer.Handle("/metrics", promhttp.Handler())
 }
 
+func registerVersion(httpServer *http.ServeMux, config *Config) {
+	httpServer.HandleFunc("/version", func(writer http.ResponseWriter, request *http.Request) {
+		version, err := json.Marshal(config.Version)
+		if err == nil {
+			writer.Header().Set("Content-Type", "application/json")
+			writer.WriteHeader(http.StatusOK)
+			writer.Write(version)
+		} else {
+			logrus.Error(err.Error())
+			http.Error(writer, "Something went a bit wrong here!", 500)
```

Comment:
This block can be simplified.

```
if err != nil {
  writer.WriteHeader(http.StatusInternalServerError)
  return
}

writer.WriteHeader(http.StatusOK)
writer.Write(version)
```

Reply:
- Is logging the issue not required?
- would I need to set Content-Type or is that automatically inferred?

Reply:
I wouldn't worry about logging it. The likelihood of this causing an error that goes unnoticed is highly unlikely. It's almost akin to serving a static file.